controllers/pipeline/pipeline_adapter_test.go

```diff
@@ -0,0 +1,258 @@
+				Application:    "application-sample",
+				ContainerImage: "",
+				Source: applicationapiv1alpha1.ComponentSource{
+					ComponentSourceUnion: applicationapiv1alpha1.ComponentSourceUnion{
+						GitSource: &applicationapiv1alpha1.GitSource{
+							URL: SampleRepoLink,
+						},
+					},
+				},
+			},
+		}
+		Expect(k8sClient.Create(ctx, hasComp)).Should(Succeed())
+	})
+
+	BeforeEach(func() {
+		testpipelineRunBuild = &tektonv1beta1.PipelineRun{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "pipelinerun-build-sample",
+				Namespace: "default",
+				Labels: map[string]string{
+					"pipelines.appstudio.openshift.io/type":  "build",
+					"pipelines.openshift.io/used-by":         "build-cloud",
+					"pipelines.openshift.io/runtime":         "nodejs",
+					"pipelines.openshift.io/strategy":        "s2i",
+					"build.appstudio.openshift.io/component": "component-sample",
+				},
+				Annotations: map[string]string{
+					"appstudio.redhat.com/updateComponentOnSuccess": "false",
+				},
+			},
+			Spec: tektonv1beta1.PipelineRunSpec{
+				PipelineRef: &tektonv1beta1.PipelineRef{
+					Name:   "build-pipeline-pass",
+					Bundle: "quay.io/kpavic/test-bundle:build-pipeline-pass",
+				},
+				Params: []tektonv1beta1.Param{
+					{
+						Name: "output-image",
+						Value: tektonv1beta1.ArrayOrString{
+							Type:      "string",
+							StringVal: "quay.io/redhat-appstudio/sample-image",
+						},
+					},
+				},
+			},
+			// 	Status: tektonv1beta1.PipelineRunStatus{
+			// 		PipelineRunStatusFields: tektonv1beta1.PipelineRunStatusFields{
+			// 			TaskRuns: map[string]*tektonv1beta1.PipelineRunTaskRunStatus{
+			// 				"index1": &tektonv1beta1.PipelineRunTaskRunStatus{
+			// 					PipelineTaskName: "build-container",
+			// 					Status: &tektonv1beta1.TaskRunStatus{
+			// 						TaskRunStatusFields: tektonv1beta1.TaskRunStatusFields{
+			// 							TaskRunResults: []tektonv1beta1.TaskRunResult{
+			// 								{
+			// 									Name:  "IMAGE_DIGEST",
+			// 									Value: "image_digest_value",
+			// 								},
+			// 							},
+			// 						},
+			// 					},
+			// 				},
+			// 			},
+			// 		},
+			// 	},
+		}
+		Expect(k8sClient.Create(ctx, testpipelineRunBuild)).Should(Succeed())
+		testpipelineRunBuild.Status = tektonv1beta1.PipelineRunStatus{
+			PipelineRunStatusFields: tektonv1beta1.PipelineRunStatusFields{
+				TaskRuns: map[string]*tektonv1beta1.PipelineRunTaskRunStatus{
+					"index1": &tektonv1beta1.PipelineRunTaskRunStatus{
+						PipelineTaskName: "build-container",
+						Status: &tektonv1beta1.TaskRunStatus{
+							TaskRunStatusFields: tektonv1beta1.TaskRunStatusFields{
+								TaskRunResults: []tektonv1beta1.TaskRunResult{
+									{
+										Name:  "IMAGE_DIGEST",
+										Value: "image_digest_value",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+		Expect(k8sClient.Status().Update(ctx, testpipelineRunBuild)).Should(Succeed())
+
+		Eventually(func() error {
```

Comment:
By changing this Eventually block to also wait for the `testpipelineRunBuild` status to update I managed to make the tests pass reliably. This is what I changed it to:
```
Eventually(func() bool {
			err := k8sClient.Get(ctx, types.NamespacedName{
				Name:      testpipelineRunBuild.Name,
				Namespace: "default",
			}, testpipelineRunBuild)
			return err == nil && len(testpipelineRunBuild.Status.TaskRuns) > 0
		}, time.Second*10).Should(BeTrue())
```

Reply:
Done